### PR TITLE
move Telemetry API to alpha

### PR DESCRIPTION
**Pull request by zirain**
Please provide a description for what this PR is for.

xref: https://github.com/istio/enhancements/pull/145

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Ambient
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [x] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/tasks/observability/telemetry/index.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ test: no
 status: Experimental
```
Comment (dhawton) [inline content/en/docs/tasks/observability/telemetry/index.md:8]:
```suggestion
status: Alpha
```

**Review by dhawton — CHANGES_REQUESTED** [review]

File: content/en/docs/tasks/observability/telemetry/index.md (hunk 1 above)
Comment (zirain) [reply content/en/docs/tasks/observability/telemetry/index.md]: done

**Review by dhawton — APPROVED** [review]

**Comment by craigbox** [comment]
Not even Beta yet? 🤣

**Pull request merged**